### PR TITLE
feat: CHK-2162 update policy for handle wallet notifications

### PR DESCRIPTION
**Pull request by infantesimone**
### List of changes

- update wallet policy notifications
### Motivation and context

The goal of this PR is to add policy to handle wallet notification according to:
- https://developer.nexi.it/it/api/notifica
- https://github.com/pagopa/pagopa-wallet-service/pull/62

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl [hunk 1]
```diff
@@ -1,15 +1,54 @@
...
+                }</set-body>
+          </return-response>
+        </when>
+      </choose>  
+      <set-body>
+          @{
+            JObject requestBody = (JObject)context.Variables["npgNotificationRequestBody"];
+            JObject operation = (JObject)requestBody["operation"];
+            string operationResult = (string)operation["operationResult"];
+            string operationId = (string)operation["operationId"];
```
Comment (Fernando-Granato) [inline src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl:28]: We can eliminate this

File: src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl [hunk 2]
```diff
@@ -1,15 +1,54 @@
...
+                DateTime npgDateTime = DateTime.Parse(operationTime.Replace(" ","T"));
+                TimeZoneInfo zone = TimeZoneInfo.FindSystemTimeZoneById("Central European Standard Time");
+                DateTime utcDateTime = TimeZoneInfo.ConvertTimeToUtc(npgDateTime, zone);
+                DateTimeOffset dateTimeOffset = new DateTimeOffset(utcDateTime);
+                timestampOperation = dateTimeOffset.ToString("o");
+            }
+            JObject request = new JObject();
+            request["timestampOperation"] = timestampOperation;
+            request["operationResult"] = operationResult;
+            request["operationId"] = operationId;
```
Comment (Fernando-Granato) [inline src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl:41]: We can eliminate this

**Review by Fernando-Granato — COMMENTED** [review]
A few small comments

**Review by Fernando-Granato — COMMENTED** [review]
You should configure the sessionToken as the Authorization header of the request, don't you think?

File: src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl (hunk 1 above)
Comment (infantesimone) [reply src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl:28]: for now we will use it as the npg notification ID

File: src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl (hunk 2 above)
Comment (infantesimone) [reply src/domains/wallet-app/api/npg-notifications/v1/_base_policy.xml.tpl:41]: ditto

**Comment by infantesimone** [comment]
> You should configure the sessionToken as the Authorization header of the request, don't you think?

correct!

**Pull request merged**